packages/react-switch/src/components/Switch/useSwitchStyles.ts

```diff
@@ -1,297 +1,231 @@
     },
 
-    ':after': {
-      position: 'absolute',
-      top: '0px',
-      left: '0px',
-      bottom: '0px',
-      right: '0px',
-      ...shorthands.borderRadius(tokens.borderRadiusCircular),
-      content: "''",
-      opacity: 'var(--switch-checked-opacity)',
+    ':active': {
+      [`& .${switchThumbClassName}`]: {
+        backgroundColor: tokens.colorNeutralForegroundDisabled,
+      },
+
+      [`& .${switchTrackClassName}`]: {
+        backgroundColor: tokens.colorNeutralBackgroundDisabled,
+        ...shorthands.borderColor(tokens.colorTransparentStrokeDisabled),
+      },
     },
   },
 
-  unchecked: {
-    ':before': {
-      backgroundColor: tokens.colorNeutralStrokeAccessible,
+  uncheckedDisabled: {
+    cursor: 'default',
+
+    [`& .${switchThumbClassName}`]: {
+      backgroundColor: tokens.colorNeutralForegroundDisabled,
     },
-  },
 
-  checked: {
-    ':after': {
-      backgroundColor: tokens.colorNeutralForegroundOnBrand,
+    [`& .${switchTrackClassName}`]: {
+      ...shorthands.borderColor(tokens.colorNeutralStrokeDisabled),
     },
-  },
 
-  disabledUnchecked: {
-    ':before': {
-      ...shorthands.border('1px', 'solid', tokens.colorNeutralForegroundDisabled),
-      backgroundColor: tokens.colorNeutralBackground1,
+    ':hover': {
+      [`& .${switchThumbClassName}`]: {
+        backgroundColor: tokens.colorNeutralForegroundDisabled,
+      },
+
+      [`& .${switchTrackClassName}`]: {
+        ...shorthands.borderColor(tokens.colorNeutralStrokeDisabled),
+      },
     },
-  },
 
-  disabledChecked: {
-    ':after': {
-      backgroundColor: tokens.colorNeutralForegroundDisabled,
+    ':active': {
+      [`& .${switchThumbClassName}`]: {
+        backgroundColor: tokens.colorNeutralForegroundDisabled,
+      },
+
+      [`& .${switchTrackClassName}`]: {
+        ...shorthands.borderColor(tokens.colorNeutralStrokeDisabled),
+      },
     },
   },
 });
 
-/**
- * Styles for the activeRail slot
- */
-const useActiveRailStyles = makeStyles({
-  activeRail: {
+const useInputStyles = makeStyles({
+  base: {
+    boxSizing: 'border-box',
+    cursor: 'inherit',
+    height: '100%',
+    left: 0,
+    ...shorthands.margin(0),
+    opacity: 0,
     position: 'absolute',
-    left: 'calc(var(--switch-thumb-size) * .7)',
-    right: 'calc(var(--switch-thumb-size) * .7)',
+    top: 0,
+    width: '100%',
+    zIndex: 1,
   },
 });
 
-/**
- * Styles for the hidden input slot
- */
-const useInputStyle = makeStyles({
-  input: {
-    opacity: 0,
-    position: 'absolute',
-    ...shorthands.padding(0),
-    ...shorthands.margin(0),
-    width: '100%',
-    height: '100%',
-    touchAction: 'none',
-    pointerEvents: 'none',
+const useThumbStyles = makeStyles({
+  base: {
+    ...shorthands.borderRadius(tokens.borderRadiusCircular),
+    boxSizing: 'border-box',
+    height: '14px',
+    transitionDuration: '200ms',
+    transitionTimingFunction: 'cubic-bezier(0.33, 0, 0.67, 1)',
+    transitionProperty: 'all',
```

Comment:
Does this need to be "all"? In my experience using "all" can have noticeable negative impacts on performance so it's better to list only the properties that need transitioning.

Reply:
I can change this for sure, I was just using all for convenience.

Reply:
This has been changed.